package/harvester-os/files/usr/sbin/harv-install

```diff
@@ -59,6 +59,9 @@ get_iso()
         get_url ${COS_INSTALL_ISO_URL} ${TEMP_FILE}
         ISO_DEVICE=$(losetup --show -f $TEMP_FILE)
         mount -o ro ${ISO_DEVICE} ${ISOMNT}
+    else
+        echo "Failed to read iso_url, it must be set in automatic install mode"
+        return 1
```

Comment:
get_iso is also called in ISO installation mode.
So this breaks the ISO installation.

Reply:
from source code:
get_iso --> get_url, which either "curl remote" or " cp local"

I guess:  for ISO install, it "cp local"
 in both cases, get_iso is called, and "$COS_INSTALL_ISO_URL" should not be empty normally. 

Will do an ISO install to check.

```
get_url()
{
    FROM=$1
    TO=$2
    case $FROM in
        ftp*|http*|tftp*)
            n=0
            attempts=5
            until [ "$n" -ge "$attempts" ]
..
                curl -o $TO -fL ${FROM} && break
     ..
        *)
            cp -f $FROM $TO
            ;;
    esac
}
```

Reply:
Verified, the ISO install failed. @bk201  

the "$COS_INSTALL_ISO_URL" is NULL when ISO install.

needs further check, thanks.

Reply:
@bk201 Please take a look of the latest commit, thanks.

https://github.com/harvester/harvester-installer/pull/163/commits/52a21c58089d37e2a23932cbfee629d0d79c03bf

Reply:
@w13915984028 The fix looks nice. But do you think it's better to put the checks in the validation code for consistency? Since we do most of the configuration validation there, other contributors might be easier to know what validations are done.

I did test my suggestion in https://github.com/harvester/harvester/issues/1439#issuecomment-965068891 and it works.

Reply:
> @w13915984028 The fix looks nice. But do you think it's better to put the checks in the validation code for consistency? Since we do most of the configuration validation there, other contributors might be easier to know what validations are done.
> 
> I did test my suggestion in [harvester/harvester#1439 (comment)](https://github.com/harvester/harvester/issues/1439#issuecomment-965068891) and it works.


@bk201 I also did a quick test with your change, it really works.  
![image](https://user-images.githubusercontent.com/31133476/141281436-cd3afa46-e7dd-4f56-a0b8-e5718c9349d0.png)


Sure, it is better to put it together with other checks.

The "validateConfig" runs after "Automatic" filed is set, my previous understanding of code flow was a bit wrong. thanks.

Please submit your change.